### PR TITLE
[TT-5962] Handle rebalance and topology changes gracefully.

### PR DESCRIPTION
**Pull request by buraksezer**
This PR fixed the bug described [here](https://tyktech.atlassian.net/jira/software/c/projects/TT/boards/26?modal=detail&selectedIssue=TT-5962). Now Kafka data source handles rebalances and topology changes gracefully.

### CODE REVIEW

File: pkg/engine/datasource/kafka_datasource/kafka_consumer_group.go [hunk 1]
```diff
@@ -123,14 +124,53 @@ func (k *KafkaConsumerGroup) startConsuming(handler sarama.ConsumerGroupHandler)
...
+	// server-side rebalance happens, the consumer session will need to be
+	// recreated to get the new claims.
+	for {
+		select {
+		case <-k.ctx.Done():
+			return
+		default:
+		}
+
+		k.log.Info("KafkaConsumerGroup.consumerGroup.Consume has been called",
```
Comment (Aenimus) [inline pkg/engine/datasource/kafka_datasource/kafka_consumer_group.go:156]: I can't help but think we can do something here with an extraction, because we're logging the exact same things, but one is in an Info, and one is an Error (with an error).

Also, we log an Info as well as an Error if we do get an error. Does that make sense?

TLDR: can we extract this into a function and do EITHER a k.log.Info or an k.log.Error?

File: pkg/engine/datasource/kafka_datasource/kafka_consumer_group.go [hunk 2]
```diff
@@ -123,14 +124,53 @@ func (k *KafkaConsumerGroup) startConsuming(handler sarama.ConsumerGroupHandler)
...
 			log.String("topic", k.options.Topic),
 			log.String("groupID", k.options.GroupID),
-			log.String("clientID", k.options.ClientID),
-			log.Error(err))
+			log.String("clientID", k.options.ClientID))
+
+		// Blocking call
+		err := k.consumerGroup.Consume(k.ctx, []string{k.options.Topic}, handler)
+		if err != nil {
```
Comment (Aenimus) [inline pkg/engine/datasource/kafka_datasource/kafka_consumer_group.go:163]: Obviously not an issue, but would it be cleaner to wrap the if on the same line?

Comment (buraksezer) [reply pkg/engine/datasource/kafka_datasource/kafka_consumer_group.go:163]: I usually prefer checking the error on a different line. Because it's mostly easy to read.

**Review by Aenimus — APPROVED** [review]

**Review by kofoworola — APPROVED** [review]
